File: app/src/main/java/org/fossasia/susi/ai/device/adapters/DeviceViewHolder.java
package org.fossasia.susi.ai.device.adapters;

import android.support.v7.widget.RecyclerView;
import android.view.View;
import android.widget.TextView;
import org.fossasia.susi.ai.R;
import org.fossasia.susi.ai.device.deviceconnect.DeviceConnectPresenter;

import butterknife.BindView;
import butterknife.ButterKnife;
import butterknife.OnClick;

public class DeviceViewHolder extends RecyclerView.ViewHolder {

    protected  @BindView(R.id.speakerName)
    TextView speakerName;
    protected  @BindView(R.id.speakerSetUp)
    TextView setUp;
    protected DeviceConnectPresenter devicePresenter;

    public DeviceViewHolder(View itemView, DeviceConnectPresenter devicePresenter) {
        super(itemView);
        ButterKnife.bind(this,itemView);
        this.devicePresenter = devicePresenter;
    }

    protected  @OnClick(R.id.speakerSetUp) void onClick() {
        String SSID = speakerName.getText().toString();
        devicePresenter.connectToDevice(SSID);
    }

}

File: app/src/main/java/org/fossasia/susi/ai/device/deviceconnect/adapters/viewholders/DeviceViewHolder.kt
package org.fossasia.susi.ai.device.deviceconnect.adapters.viewholders

import android.view.View
import android.widget.TextView
import androidx.recyclerview.widget.RecyclerView
import kotterknife.bindView
import org.fossasia.susi.ai.R
import org.fossasia.susi.ai.device.deviceconnect.DeviceConnectPresenter

class DeviceViewHolder(itemView: View, private var devicePresenter: DeviceConnectPresenter) : androidx.recyclerview.widget.RecyclerView.ViewHolder(itemView) {

    val speakerName: TextView by bindView(R.id.speakerName)
    val setUp: TextView by bindView(R.id.speakerSetUp)

    // Initializing the onClick function
    init {
        setUp.setOnClickListener {
            onClick()
        }
    }

    fun onClick() {
        val ssid = speakerName.text.toString()
        devicePresenter.connectToDevice(ssid)
    }
}
